test/cql-pytest/cassandra_tests/validation/operations/select_single_column_relation_test.py

```diff
@@ -20,6 +20,33 @@ def testInvalidCollectionEqualityRelation(cql, test_keyspace):
         assert_invalid_message(cql, table, "Collection column 'd' (map<int, int>) cannot be restricted by a '=' relation",
                              "SELECT * FROM %s WHERE a = 0 AND d=?", {0: 0})
 
+def testInvalidCollectionIntegerOperandEqualityRelation(cql, test_keyspace):
```

Comment:
The tests in test/cql-pytest/cassandra_tests/ are all tests translated - and otherwise unmodified - from Cassandra's unit tests.
So if you want to add a new test, please add it in test/cql-pytest but not in the "cassandra_tests" subdirectory.

Reply:
Please add a comment in front (or if you prefer, Python doc-string...) of every test explaining what its goal is, and referring to an issue number of it reproduces an issue. I thought the name of this test is test was self-explanatory, but then I saw code which did other things, so I started to wonder.

Reply:
By the way, I assume you did, but if you didn't, please remember to run the new tests also with `test/cql-pytest/run-cassandra`, to verify that these tests also pass in Cassandra - i.e., the behavior you are testing for is indeed compatible with Cassandra and not our mistaken understanding of Cassandra's behavior.

Reply:
> The tests in test/cql-pytest/cassandra_tests/ are all tests translated - and otherwise unmodified - from Cassandra's unit tests. So if you want to add a new test, please add it in test/cql-pytest but not in the "cassandra_tests" subdirectory.
Thanks, will fix it: move test to test/cql-pytest

Reply:
> By the way, I assume you did, but if you didn't, please remember to run the new tests also with `test/cql-pytest/run-cassandra`, to verify that these tests also pass in Cassandra - i.e., the behavior you are testing for is indeed compatible with Cassandra and not our mistaken understanding of Cassandra's behavior.

Yes I did but also thought these tests should be run from general `test.py`? Or I should always run cassandra tests separately?

Reply:
test.py is a tool to run all types of tests of Scylla, including those in test/cql-pytest but also many other types of tests (C++ unit tests, etc.). It's always good to run it when you finish development (and our CI system will run automatically before a patch is committed). 

But in this case, you are aiming for (and claiming) Cassandra compatibility, so how do you know your test itself isn't buggy? I.e., you are checking that in some case you get the error message XYZ - but maybe Cassandra doesn't actually print this error message in this case and you misunderstood that it does? This is why the design principle behind cql-pytest is that it uses only official CQL requests, so exactly the same test can be run not just on Scylla, but also on Cassandra. When you write the test, you should make sure it passes on Cassandra (which means the test itself is correct), and then that it also passes on Scylla (which means Scylla isn't buggy). We have a convenient script to start Cassandra in a temporary directory and run the tests on it - `test/cql-pytest/run-cassandra`. For example, `test/cql-pytest/run-cassandra test_file.py::test_function |& less` runs a single test function and shows the result (and the Cassandra log).

Note that test.py does not - and cannot - run Cassandra, so it cannot replace the `test/cql-pytest/run-cassandra`. It may replace `test/cql-pytest/run` (which runs a test on Scylla), if you wish, though I still prefer the more powerful and faster `test/cql-pytest/run`.

Reply:
Huge thanks for your explanation, it explained me many things I met with tests. Will check tests again after fixing